cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerTemplateInstallationChecker.java

```diff
@@ -39,48 +34,23 @@ protected boolean doStatusCheck(ClouderaManagerCommandPollerObject pollerObject,
         } else if (apiCommand.getSuccess()) {
             return true;
         } else {
-            List<String> errorReasons = new ArrayList<>();
-            digForFailureCause(apiCommand, errorReasons, commandsResourceApi);
-            String msg = "Cluster template install failed: " + errorReasons;
-            LOGGER.info(msg);
-            throw new ClouderaManagerOperationFailedException(msg);
+            fail("", apiCommand, commandsResourceApi);
         }
+        return false;
     }
 
-    private Predicate<ApiCommand> commandFailed() {
-        return cmd -> !cmd.getSuccess();
-    }
-
-    private Function<ApiCommand, ApiCommand> readCommand(CommandsResourceApi commandsResourceApi) {
-        return cmd -> {
-            try {
-                return commandsResourceApi.readCommand(cmd.getId());
-            } catch (ApiException e) {
-                LOGGER.debug("Failed to read command. id [{}]", cmd.getId(), e);
-                return new ApiCommand();
-            }
-        };
-    }
-
-    private void digForFailureCause(ApiCommand apiCommand, List<String> errorReasons, CommandsResourceApi commandsResourceApi) {
-        List<String> childErrors = new LinkedList<>();
-        Optional.ofNullable(apiCommand.getChildren()).map(ApiCommandList::getItems).orElse(List.of()).stream()
-                .filter(commandFailed())
-                .map(readCommand(commandsResourceApi))
-                .forEach(cmd -> digForFailureCause(cmd, childErrors, commandsResourceApi));
-
-        if (childErrors.isEmpty() && StringUtils.isNotEmpty(apiCommand.getResultMessage())) {
-            String reason = String.format("Command [%s], with id [%.0f] failed: %s", apiCommand.getName(), apiCommand.getId(), apiCommand.getResultMessage());
-            errorReasons.add(reason);
+    @Override
+    public void handleTimeout(ClouderaManagerCommandPollerObject pollerObject) {
+        String msg = "Installation of CDP with Cloudera Manager has timed out (command id: " + pollerObject.getId() + ").";
+        try {
+            CommandsResourceApi commandsResourceApi = clouderaManagerApiPojoFactory.getCommandsResourceApi(pollerObject.getApiClient());
+            ApiCommand apiCommand = commandsResourceApi.readCommand(pollerObject.getId());
+            fail(msg, apiCommand, commandsResourceApi);
+        } catch (ApiException e) {
+            LOGGER.info("Cloudera Manager had run into a timeout and, we were unable to determine the failure reason", e);
```

Comment:
The comma seems misplaced, should come before `and`.

Reply:
Thanks, moved.